Update error responses

RaccoonHowell opened the pull request:
## Summary
* Routine Change
* :exclamation: Breaking Change
* :robot: Operational or Infrastructure Change
* :sparkles: New Feature
* :warning: Potential issues that might be caused by this change

**I added in all fields from this page: [https://nhsd-confluence.digital.nhs.uk/pages/viewpage.action?spaceKey=APM&title=Handling+errors]. I don't think the expression property is relevant to 5XX errors so put a note on line  691 - it seems to show up ok in the swagger preview if I just comment out that property in the example but not sure if this is good practice or if the property should be removed from the schema and added only to 4xx errors?

It is quite lengthy now so also wondered if we could merge 4XX and 5XX and just have one response/schema/example to cover all error responses?**

## Reviews Required
* [x] Dev
* [ ] Test
* [ ] Tech Author
* [ ] Product Owner


## Review Checklist
:information_source: This section is to be filled in by the **reviewer**.

* [ ] I have reviewed the changes in this PR and they fill all or part of the acceptance criteria of the ticket, and the code is in a mergeable state.
* [ ] If there were infrastructure, operational, or build changes, I have made sure there is sufficient evidence that the changes will work.
* [ ] I have ensured the changelog has been updated by the submitter, if necessary.

st3v3nhunt commented on specification/gp-connect-user-permissions.yaml:
```diff
@@ -193,6 +193,7 @@ components:
         returned.
       required:
         - resourceType
+        - id
```
See comment about `id` - I think it should probably be removed.

st3v3nhunt commented on specification/gp-connect-user-permissions.yaml:
```diff
@@ -220,8 +225,19 @@ components:
             required:
+              - severity
               - code
+              - details
```
Looking at the info in Confluence and the signing service error I don't think `details` is required.

st3v3nhunt commented on specification/gp-connect-user-permissions.yaml:
```diff
@@ -203,6 +204,10 @@ components:
             in non-FHIR APIs for consistency across the platform.
           enum:
             - OperationOutcome
+        id:
```
As `id` is for an apigee error I think it should be removed from the spec - the signing service doesn't include it either.

st3v3nhunt commented on specification/gp-connect-user-permissions.yaml:
```diff
@@ -608,19 +657,40 @@ components:
     InvalidRequestError:
       value:
         resourceType: OperationOutcome
+        id: rrt-2959959087381887325-c-geu2-24001-82918062-1
```
Please remove `id` from the example if removed from the schema.

st3v3nhunt commented on specification/gp-connect-user-permissions.yaml:
```diff
@@ -608,19 +657,40 @@ components:
     ServerError:
       value:
         resourceType: OperationOutcome
+        id: rrt-2959959087381887325-c-geu2-24001-82918062-2
```
As above - please remove.

st3v3nhunt reviewed (changes requested):
Looks good, just a couple of changes please - mostly around `id` as I don't think it should be included.

It might be worth added another example with just the required fields added to the object, although that could be a separate PR.

RaccoonHowell commented:
Deleted id property and removed details from the required section. Changed error responses content to application/fhir+json as per this API that is also non-FHIR but uses FHIR format error responses [https://github.com/NHSDigital/health-research-data-catalogue/blob/master/specification/health-research-data-catalogue.yaml](url)

st3v3nhunt reviewed (approved):
LGTM 👍 

Good spot on the content type 👀
